flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/FMC/A32NX_FMCMainDisplay.js

```diff
@@ -1626,41 +1626,29 @@ class FMCMainDisplay extends BaseAirliners {
         return this._routeTripFuelWeight;
     }
 
-    getOrSelectVORsByIdent(ident, callback) {
-        this.dataManager.GetVORsByIdent(ident).then((navaids) => {
-            if (!navaids || navaids.length === 0) {
-                this.addNewMessage(NXSystemMessages.notInDatabase);
+    _getOrSelectWaypoints(getter, ident, callback) {
```

Comment:
re-factored common logic into one function here. Shifted responsibility for generating the not in database scratchpad message to the caller for VOR and NDB, as it already was for the catch-all waypoint function.

Reply:
**thought**: I'm in doubt about having this function here. It feels like filtering is a concern of the `FMCDataManager`. Of course some other things this function does should be part of `FMCMainDisplay`.

I guess it's the somewhat "interesting" style of Asobo JavaScript coding that makes it hard to get this very clean right now.

Reply:
I'll take a look and refactor it a little more. You're right that the data manager shouldn't send duplicates in the first place and it makes sense to filter then there for the benefit of other systems too.